### PR TITLE
Get by area

### PR DESCRIPTION
**Pull request by festinalente91**
added get by area.
I tested get items include neighbor by area with query lat,lng,zoom

### CODE REVIEW

File: src/server/items/controllers.js [hunk 1]
```diff
@@ -1,27 +1,64 @@
...
-      items.push(data.value);
-    }).on('error', (err) => {
-      error = err;
-      return cb(new APIError(err));
-    })
-    .on('close', () => {
-      if (!error) {
-        res.status(200).send({items});
-        cb();
+    if (req.query.is_area_search === 'true') {
```
Comment (ovekyc) [inline src/server/items/controllers.js]: You can deal this if statement with just 
lat lng zoom

File: src/server/items/controllers.js [hunk 2]
```diff
@@ -1,27 +1,64 @@
...
+      const {lat, lng, zoom} = req.query;
+      const precision = Math.floor((Number(zoom) + 1) / 3);
```
Comment (ovekyc) [inline src/server/items/controllers.js]: Please make module for precision

File: src/server/items/routes.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import express from 'express';
-import ItemContollers from './controllers';
+import ItemControllers from './controllers';
 
 const router = express.Router();
 
 /**
- * @api {get} /items Get All items
+ * @api {get} /items?is_area_search=true&lat=30.565398&lng=126.9907941&zoom=21 Get All items
```
Comment (hewonjeong) [inline src/server/items/routes.js]: we need 'is_area_search'?
how bout removing this?

File: src/server/items/controllers.js [hunk 4]
```diff
@@ -1,27 +1,64 @@
...
       }
-    });
+      Promise.all(promises).then(() => {
+        res.status(200).send({
+          items
+        });
+        cb();
+      }).catch((err) => {
+        return cb(new APIError(err));
+      });
```
Comment (ovekyc) [inline src/server/items/controllers.js]: You can return at this point and else bracket

File: src/test/node/items/controllers.js [hunk 5]
```diff
@@ -115,6 +115,57 @@ test('get a item from database', t => {
...
+      { id: `item-${ALIVE}-${neighbors[3]}-${reversedTime}`},
+      { id: `item-${ALIVE}-${neighbors[4]}-${reversedTime}`},
+      { id: `item-${ALIVE}-${neighbors[5]}-${reversedTime}`},
+      { id: `item-${ALIVE}-${neighbors[6]}-${reversedTime}`},
+      { id: `item-${ALIVE}-${neighbors[7]}-${reversedTime}`}
+    ]
+  };
+  const req = httpMocks.createRequest({
+    method: 'GET',
+    url: '/items?is_area_search=true&lat=30.565398&lng=126.9907941&zoom=21'
```
Comment (ovekyc) [inline src/test/node/items/controllers.js]: same here

File: src/server/items/routes.js (hunk 3 above)
Comment (ovekyc) [inline src/server/items/routes.js]: same here

File: src/test/node/items/controllers.js [hunk 6]
```diff
@@ -115,6 +115,57 @@ test('get a item from database', t => {
...
+    for (let i = 0; i < expected.items.length; i = i + 1) {
+      const tempValue = JSON.parse(JSON.stringify(itemRedSelo));
+      tempValue.ref = expected.items[i].id;
+      ops.push({ type: 'put', key: expected.items[i].id, value: tempValue });
+    }
+    testDB.batch(ops, (err) => {
+      if (err) {
+        t.end(err);
+      }
+      ItemController.getAll(req, res, () => {
```
Comment (ovekyc) [inline src/test/node/items/controllers.js:154]: same here

File: src/server/items/controllers.js [hunk 7]
```diff
@@ -1,27 +1,64 @@
 import db from '../database';
 import {APIError} from '../ErrorHandler';
-import {KeyMaker, DEFAULT_PRECISON} from './models';
+import {KeyMaker, KeyUtils, ALIVE, DEFAULT_PRECISON} from './models';
 
 export default {
   getAll: (req, res, cb) => {
```
Comment (ovekyc) [inline src/server/items/controllers.js:7]: please change function name

Comment (festinalente91) [reply src/server/items/controllers.js:7]: Any recommendations? haha
I thought basically this function is used for querying all items,
and if you want to add some conditions you can add querystring for that. :)

File: src/server/items/controllers.js (hunk 1 above)
Comment (festinalente91) [reply src/server/items/controllers.js]: right:) I see

File: src/server/items/controllers.js (hunk 2 above)
Comment (festinalente91) [reply src/server/items/controllers.js]: good point

**Comment by ovekyc** [comment]
LGTM

File: src/server/items/routes.js (hunk 3 above)
Comment (festinalente91) [reply src/server/items/routes.js]: removed !

File: src/server/items/controllers.js (hunk 4 above)
Comment (festinalente91) [reply src/server/items/controllers.js]: updated

**Comment by hewonjeong** [comment]
LGTM

**Pull request merged**